dune/grid/CpGrid.hpp

```diff
@@ -819,6 +820,149 @@ namespace Dune
             }
             return zz/nv;
         }
+
+        const Vector faceCenterEcl(int cell_index, int faceTag) const
+        {
+            // This method is an alternative to the method faceCentroid(...).
+            // The face center is computed as a raw average of cell corners.
+            // For faulted cells this gives different results then average of face nodes
+            // that seems to agree more with eclipse.
+            // This assumes the cell nodes are ordered
+            // 0--1
+            // |  |
+            // 2--3
+            //   4--5
+            //   |  |
+            //   6--7
```

Comment:
Assuming a right-handed `i-j` system the eclipse ordering is in my opinion
```txt
2----3
|    |
0----1
```

and

```txt
6----7
|    |
4----5
```

i.e. the shift in global index when going 0 &rarr; 2 is `nx` - and not `-nx` which the supplied drawing might indicate.

I have not looked at the actual implementation; this is probably just a detail in the comment.